lib/scidata/imdb_reviews.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Scidata.IMDBReviews do
+  @moduledoc """
+  Module for downloading the [Large Movie Review Dataset](https://ai.stanford.edu/~amaas/data/sentiment/).
+  """
+
+  @base_url "http://ai.stanford.edu/~amaas/data/sentiment/"
+  @dataset_file "aclImdb_v1.tar.gz"
+
+  alias Scidata.Utils
+
+  @doc """
+  Downloads the IMDB reviews training dataset or fetches it locally.
+
+  `example_types` specifies which examples in the dataset should be returned
+  according to each example's label: `:pos` for positive examples, `:neg` for
+  negative examples, and `:unsup` for unlabeled examples.
+  """
+  def download(
+        example_types \\ [:pos, :neg],
+        opts \\ []
+      ) do
+    download_dataset(example_types, :train, opts)
+  end
+
+  @doc """
+  Downloads the IMDB reviews test dataset or fetches it locally.
+
+  `example_types` is the same argument in `download/2` but excludes `:unsup`
+  because all unlabeled examples are in the training set.
+  """
+  def download_test(
+        example_types \\ [:pos, :neg],
+        opts \\ []
+      ) do
+    download_dataset(example_types, :test, opts)
+  end
+
+  defp download_dataset(example_types, dataset_type, opts) do
+    transform_inputs = opts[:transform_inputs] || (& &1)
+    transform_labels = opts[:transform_labels] || (& &1)
+
+    files = Utils.get!(@base_url <> @dataset_file).body
+
+    {inputs, labels} =
+      files
+      |> Enum.filter(fn {fname, _} ->
+        file_match?(fname, dataset_type, example_types)
+      end)
+      |> Enum.reduce(
+        {[], []},
+        fn {fname, contents}, {inputs, labels} ->
+          {[contents | inputs], [get_label(fname) | labels]}
+        end
+      )
```

Comment:
You can also try this for a single pass on the data :)

```suggestion
      for {fname, contents} <- files,
          file_match?(fname, dataset_type, example_types),
          reduce: {[], []} do
        {inputs, labels} ->
          {[contents | inputs], [get_label(fname) | labels]}
      end)
```